packages/atomic/src/components/insight/atomic-insight-refine-modal/atomic-insight-refine-modal.tsx

```diff
@@ -45,17 +40,14 @@ export class AtomicInsightRefineModal
   public error!: Error;
 
   @State()
-  public loadingDimensions = true;
+  public interfaceDimensions?: DOMRect;
 
   @Prop({mutable: true}) openButton?: HTMLElement;
 
   @Prop({reflect: true, mutable: true}) isOpen = false;
-
-  private interfaceDimensions?: InsightInterfaceDimensions;
   private facetManager!: InsightFacetManager;
   private resizeObserver?: ResizeObserver;
-  private debouncedUpdateDimensions = debounce(this.updateDimensions, 500);
-  private scrollCallback = () => this.debouncedUpdateDimensions();
+  private scrollCallback = () => this.updateDimensions();
```

Comment:
Was the debounce here because it was laggy? How does it look now?

Reply:
I assumed it was due to waiting for the event handler to be present, but actually it seems to be to prevent refreshing too often. Will fix.

https://github.com/coveo/ui-kit/pull/2278#discussion_r936745666

Reply:
The debounce makes the user experience pretty horrible in full screen mode when scrolling

Reply:
I'll try switching to [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) instead of an event, and I'll remove the debounce.